packages/builder/src/builderStore/store/frontend.js

```diff
@@ -1246,6 +1246,15 @@ export const getFrontendStore = () => {
           const settings = getComponentSettings(component._component)
           const updatedSetting = settings.find(setting => setting.key === name)
 
+          const resetFields = settings.filter(
+            setting => name === setting.resetOn
+          )
+          resetFields?.forEach(setting => {
+            if (component[setting.key]) {
```

Comment:
I would probably just always set this to null, and remove the truthy check - there might be values like `false` or `0` or `""` which we should reset, but we would skip them with this check here since they are loosely falsey.